test/with_dummyserver/test_poolmanager.py

```diff
@@ -378,6 +379,60 @@ def test_top_level_request_with_body(self):
         assert r.status == 200
         assert r.data == b"test"
 
+    def test_top_level_request_with_preload_content(self):
+        r = request("GET", f"{self.base_url}/echo", preload_content=False)
+        assert r.status == 200
+        assert r.connection is not None
+        r.data
+        assert r.connection is None
+
+    def test_top_level_request_with_redirect(self):
+        r = request(
+            "GET",
+            f"{self.base_url}/redirect",
+            fields={"target": f"{self.base_url}/"},
+            redirect=False,
+        )
+
+        assert r.status == 303
+
+        r = request(
+            "GET",
+            f"{self.base_url}/redirect",
+            fields={"target": f"{self.base_url}/"},
+            redirect=True,
+        )
+
+        assert r.status == 200
+        assert r.data == b"Dummy server!"
+
+    def test_top_level_request_with_retries(self):
+        r = request("GET", f"{self.base_url}/redirect", retries=False)
+        assert r.status == 303
+
+        r = request("GET", f"{self.base_url}/redirect", retries=3)
+        assert r.status == 200
+
+    def test_top_level_request_with_timeout(self):
+        with mock.patch("urllib3.poolmanager.RequestMethods.request") as mockRequest:
```

Comment:
Is mocking with context manager okay here? Like I can do this with a mock decorator well.

Reply:
IMO context manager is preferred. :D

Reply:
Okey-dok! :)